SingularityService/src/main/java/com/hubspot/singularity/resources/TaskResource.java

```diff
@@ -634,19 +643,22 @@ public Response downloadFileOverProxy(
             httpPrefix, slaveHostname, httpPort);
 
     try {
-      final InputStream responseStream = httpClient.prepareGet(url)
-          .addQueryParameter("path", fileFullPath)
-          .execute()
-          .get()
-          .getResponseBodyAsStream();
+      PerRequestConfig unlimitedTimeout = new PerRequestConfig();
+      unlimitedTimeout.setRequestTimeoutInMs(-1);
```

Comment:
The default 60 second timeout on our AsyncHttpClient is insufficient for large file downloads.